app/models/enterprise.rb

```diff
@@ -96,12 +96,10 @@ class Enterprise < ActiveRecord::Base
   scope :not_ready_for_checkout, lambda {
     # When ready_for_checkout is empty, ActiveRecord generates the SQL:
     # id NOT IN (NULL)
-    # I would have expected this to return all rows, but instead it returns none. To
-    # work around this, we use the "OR ?=0" clause to return all rows when there are
-    # no enterprises ready for checkout.
-    where('id NOT IN (?) OR ?=0',
-          Enterprise.ready_for_checkout,
-          Enterprise.ready_for_checkout.count)
+    # I would have expected this to return all rows, but instead it returns none. But we want to
+    # return all rows when there are no enterprises ready for checkout.
+    ready_enterprises = Enterprise.ready_for_checkout
+    ready_enterprises.present? ? where("id NOT IN (?)", ready_enterprises) : where("TRUE")
```

Comment:
This reduces the number of queries only by 1 for each call.

Reply:
I think the comment above needs to be adapted to the new version.

also, quite a complicated thing happening, so I'd prefer to see the condition in multiple lines:
```suggestion
    if ready_enterprises.present?
       where("id NOT IN (?)", ready_enterprises)
    else
       where("TRUE")
    end
```

Reply:
Thanks, @luisramos0! Yes, I agree. Changed here: https://github.com/openfoodfoundation/openfoodnetwork/pull/3802/commits/759316834154ed9c463c0ef06dcba469a56f2c72